backend/generate-db.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"os"
+)
+
+type Course struct {
+	ID   uint   `gorm:"primaryKey" json:"-"`
+	Code string `gorm:"unique,not null" json:"code"`
+	Name string `gorm:"not null" json:"name"`
+}
+
+type Tutor struct {
+	ID       uint   `gorm:"primaryKey" json:"-"`
+	Username string `gorm:"unique,not null" json:"username"`
+}
+
+type Tutoring struct {
+	TutorID  uint   `json:"-"`
+	Tutor    Tutor  `gorm:"foreignKey:TutorID" json:"tutor"`
+	CourseID uint   `json:"-"`
+	Course   Course `gorm:"foreignKey:CourseID" json:"course"`
+}
+
+func main() {
+	err := os.Remove("database.db") //remove old database, if needed
+	if err == nil {
+		fmt.Println("Deleting old database")
```

Comment:
Minor nitpick - for clarity, this should be deleted since it's after `os.Remove`. I'd recommend just putting this before the remove call and ignoring the error.